File: src/types/JobPosting.ts
export interface JobPosting {
  id: string;
  title: string;
  company: string;
  location: string;
  description: string;
  requirements: string[];
  salary?: string;
  contactEmail: string;
  createdAt: string;
  createdBy: string;
  type: 'full-time' | 'part-time' | 'contract' | 'freelance';
  workMode: 'remote' | 'hybrid' | 'on-site';
} 

File: WebApplication1/WebApplication1/Models/JobPosting.cs
using System;
using System.Collections.Generic;

namespace WebApplication1.Models
{
    public class JobPosting
    {
        public string Id { get; set; }
        public string Title { get; set; }
        public string Company { get; set; }
        public string Location { get; set; }
        public string Description { get; set; }
        public List<string> Requirements { get; set; }
        public string Salary { get; set; }
        public string ContactEmail { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; }
        public string Type { get; set; } // full-time, part-time, contract, freelance
        public string WorkMode { get; set; } // remote, hybrid, on-site
    }
} 